### PR TITLE
refactor(dfx): `crate::lib::project` concrete error types 

### PR DESCRIPTION
**Pull request by smallstepman**
# Description

Concrete error types for `crate::lib::project::{import, network_mappings}`

In preparation to move to `dfx-core` crate.

# How Has This Been Tested?

Covered by CI.

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] ~I have edited the CHANGELOG accordingly.~
- [ ] ~I have made corresponding changes to the documentation.~

### CODE REVIEW

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
project/import can probably move into the nns/sns extensions, and we can remove the beta project import commands.  This code wouldn't move to dfx-core.

**Comment by smallstepman** [comment]
I think it's still beneficial to pursue merging this because:
- https://dfinity.atlassian.net/browse/SDK-566 mentions we should modularize `dfx`. Even if this stuff cannot go to `dfx-core`, I'd argue the functionality is enough generic to be worthy of putting into a separate crate instead of being baked into `nns` extension "forever", instead, we could put it into `dfx-extensions-core` or `dfx-project` crate.
- improves over current error messages

**Review by ericswanson-dfinity — APPROVED** [review]
🤷‍♂️ Okay, but please don't try to move this to dfx-core.

File: src/dfx/src/lib/error/project.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#[derive(Error, Debug)]
+pub enum ProjectError {
+    #[error(transparent)]
+    StructuredFileError(#[from] dfx_core::error::structured_file::StructuredFileError),
+
+    #[error(transparent)]
+    IoError(#[from] dfx_core::error::io::IoError),
+
+    // Q: ok to handle Infallible?
```
Comment (smallstepman) [inline src/dfx/src/lib/error/project.rs:11]: TODO

**Review by smallstepman — COMMENTED** [review]
figure out these TODOs before merging

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**